layers/stp.go

```diff
@@ -92,18 +82,14 @@ func (stp *STP) NextLayerType() gopacket.LayerType {
 }
 
 // Check if the priority value is correct.
-func checkPriority(prio STPPriority) (uint16, error) {
-	switch prio {
-	case STPPriorityMax:
-		return uint16(prio), nil
-	case STPPriorityHigh:
-		return uint16(prio), nil
-	case STPPriorityLow:
-		return uint16(prio), nil
-	case STPPriorityMin:
-		return uint16(prio), nil
-	default:
-		return uint16(prio), errors.New("Invalid Priority value must be one of the following:\n32768\n16384\n8192\n4096\n")
+func checkPriority(prio uint16) (uint16, error) {
+	if prio == 0 {
+		return prio, errors.New("Invalid Priority value must be in the rage <4096-61440> with an increment of 4096")
+	}
+	if prio%4096 == 0 {
+		return prio, nil
+	} else {
+		return prio, errors.New("Invalid Priority value must be in the rage <4096-61440> with an increment of 4096")
 	}
 }
```

Comment:
I've changed the check priority process since in the previous one i only check for 4 possible values whereas in reality there is 16 possible values so i decided to remove STPPrioritype since there is a lot more possible values.
1000                                000000000000
--prio--                             ------vlanID------
4bytes                                   12Bytes
16 possible values          4096 possible values